examples/src/main/java/tachyon/examples/BasicNonByteBufferOperations.java

```diff
@@ -61,17 +78,27 @@ public BasicNonByteBufferOperations(TachyonURI masterLocation, TachyonURI filePa
 
   @Override
   public Boolean call() throws Exception {
-    TachyonFS client = TachyonFS.get(mMasterLocation, new TachyonConf());
-
-    write(client, mFilePath, mWriteType, mDeleteIfExists, mLength);
-    return read(client, mFilePath, mReadType);
+    TachyonConf tachyonConf = ClientContext.getConf();
+    tachyonConf.set(Constants.MASTER_HOSTNAME, mMasterLocation.getHost());
+    tachyonConf.set(Constants.MASTER_PORT, Integer.toString(mMasterLocation.getPort()));
+    if (sReinitializer == null) {
+      ClientContext.accessReinitializer(sReinitializerAccesser);
+    }
+    sReinitializer.reinitializeWithConf(tachyonConf);
+    TachyonFileSystem tFS = TachyonFileSystem.get();
+    write(tFS, mFilePath, mWriteType, mDeleteIfExists, mLength);
+    return read(tFS, mFilePath, mReadType);
   }
 
-  private void write(TachyonFS client, TachyonURI filePath, WriteType writeType,
-      boolean deleteIfExists, int length) throws IOException {
+  private void write(TachyonFileSystem tachyonFileSystem, TachyonURI filePath,
+      TachyonStorageType writeType, boolean deleteIfExists, int length)
+          throws IOException, BlockInfoException, FileDoesNotExistException, InvalidPathException,
+          FileAlreadyExistException {
+    ClientOptions clientOptions =
+        new ClientOptions.Builder(ClientContext.getConf()).setTachyonStoreType(writeType).build();
     // If the file exists already, we will override it.
-    TachyonFile file = getOrCreate(client, filePath, deleteIfExists);
-    DataOutputStream os = new DataOutputStream(file.getOutStream(writeType));
+    TachyonFile file = getOrCreate(tachyonFileSystem, filePath, deleteIfExists, clientOptions);
+    DataOutputStream os = new DataOutputStream(new FileOutStream(file.getFileId(), clientOptions));
```

Comment:
Try not to use FileOutStream directly, perhaps `getOutStream` instead? The logic for `getOrCreate` will also need to be changed, since Tachyon files can only be written once.